cmd/registry-migrator/main.go

```diff
@@ -0,0 +1,212 @@
+// Copyright (c) 2019 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"context"
+	"flag"
+	"log"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/coreos/etcd/clientv3"
+	"github.com/coreos/etcd/clientv3/clientv3util"
+	"github.com/coreos/etcd/clientv3/snapshot"
+	"github.com/coreos/etcd/etcdserver/api/v3rpc/rpctypes"
+	"github.com/coreos/etcd/pkg/transport"
+	"go.uber.org/zap"
+	"google.golang.org/grpc/grpclog"
+)
+
+const (
+	dialTimeout       = 5 * time.Second
+	keyExistsErrorMsg = `either key already exists or revision of the old key was changed when doing migration.
+Use the "--force" flag to skip this check.
+Aborting %+v.
+`
+)
+
+func main() {
+	var (
+		oldRegistryPrefix,
+		newRegistryPrefix,
+		caFile,
+		certFile,
+		keyFile,
+		endpoints,
+		backup string
+		delete, force bool
+	)
+
+	flag.StringVar(&oldRegistryPrefix, "old-registry-prefix", "/registry/garden.sapcloud.io", "default apiserver registry prefix")
+	flag.StringVar(&newRegistryPrefix, "new-registry-prefix", "/registry-gardener", "new apiserver registry prefix")
+	flag.StringVar(&caFile, "ca", "", "path to etcd's certificate authority file")
+	flag.StringVar(&certFile, "cert", "", "path to etcd client public certificate file")
+	flag.StringVar(&keyFile, "key", "", "path to etcd client certificate key file")
+	flag.StringVar(&endpoints, "endpoints", "http://localhost:2379", "comma-separated list of etcd endpoints")
+	flag.BoolVar(&force, "force", false, "forcefully override existing keys")
+	flag.BoolVar(&delete, "delete", false, "deletes migrated keys")
+	flag.StringVar(&backup, "backup-file", "", "file to store backup to")
+
+	flag.Parse()
+
+	clientv3.SetLogger(grpclog.NewLoggerV2(os.Stderr, os.Stderr, os.Stderr))
```

Comment:
Please use `os.StdOut` for info and warning logs.
```suggestion
	clientv3.SetLogger(grpclog.NewLoggerV2(os.StdOut, os.StdOut, os.Stderr))
```